openstack/networking/v2/extensions/quotas/results.go

```diff
@@ -128,3 +133,38 @@ type QuotaDetail struct {
 	// allocated/provisioned.  This is what "quota" usually refers to.
 	Limit int `json:"limit"`
 }
+
+// UnmarshalJSON overrides the default unmarshalling function to accept
+// Reserved as a string.
+//
+// Due to a bug in Neutron, under some conditions Reserved is returned as a
+// string.
+//
+// This method is left for compatibility with unpatched versions of Neutron.
+//
+// cf. https://bugs.launchpad.net/neutron/+bug/1918565
+func (q *QuotaDetail) UnmarshalJSON(b []byte) error {
+	var quotaDetail struct {
+		Used     int             `json:"used"`
+		Reserved json.RawMessage `json:"reserved"`
```

Comment:
it's better to use interface to align with the rest gophercloud style:
see a https://github.com/gophercloud/gophercloud/blob/cd9c207e93f4f76af2c0a06c6d449ab342bfbe56/openstack/compute/v2/extensions/hypervisors/results.go#L39 for example

Reply:
@kayrus Thanks!